velox/exec/TableWriter.cpp

```diff
@@ -242,16 +243,22 @@ std::string TableWriter::createTableCommitContext(bool lastOutput) {
   // clang-format on
 }
 
-void TableWriter::updateWrittenBytes() {
-  const auto writtenBytes = dataSink_->getCompletedBytes();
-  auto lockedStats = stats_.wlock();
-  lockedStats->physicalWrittenBytes = writtenBytes;
-}
-
-void TableWriter::updateNumWrittenFiles() {
-  auto lockedStats = stats_.wlock();
-  lockedStats->addRuntimeStat(
-      "numWrittenFiles", RuntimeCounter(dataSink_->numWrittenFiles()));
+void TableWriter::updateStats(const connector::DataSink::Stats& stats) {
+  {
+    auto lockedStats = stats_.wlock();
+    lockedStats->physicalWrittenBytes = stats.completedBytes;
+    if (!closed_) {
+      // NOTE: the other stats is only set when hive data sink is closed.
+      VELOX_CHECK(stats.partitions.empty());
+      VELOX_CHECK(stats.spillStats.empty());
+      return;
+    }
+    lockedStats->addRuntimeStat(
+        "numWrittenFiles", RuntimeCounter(stats.numWrittenFiles));
+  }
+  if (!stats.spillStats.empty()) {
+    recordSpillStats(stats.spillStats);
+  }
```

Comment:
Where do we update the partitions?

Reply:
Partitions is not used here but used in table writer output.

Reply:
but `updateStats(stats);` is called by table writer output right?

Reply:
Synced offline with @kewang1024